### PR TITLE
add forecast submodule in weather to provide weather forecast

### PR DESCRIPTION
**Pull request by normcyr**

### CODE REVIEW

File: sopel/modules/weather.py [hunk 1]
```diff
@@ -151,6 +174,41 @@ def weather(bot, trigger):
...
+def forecast(bot, trigger):
+    """.forecast location - Show the weather forecast for tomorrow at the given location."""
+
+    location = trigger.group(2)
+    woeid = ''
+    if not location:
+        woeid = bot.db.get_nick_value(trigger.nick, 'woeid')
+        if not woeid:
+            return bot.msg(trigger.sender, "I don't know where you live. " +
+                           'Give me a location, like .weather London, or tell me where you live by saying .setlocation London, for example.')
```
Comment (dgw) [inline sopel/modules/weather.py]: This message should give `.forecast` instead of `.weather` as the example.

**Comment by normcyr** [comment]
Nice catch! I corrected it.

Sorry for the initial erroneous commit.

**Comment by dgw** [comment]
Rebased to squash the commits. Good to go :shipit:

**Comment by dgw** [comment]
Did the refactor I talked about. @normcyr Did I break anything you added? :grin:

**Pull request merged**